### PR TITLE
[0615] Migrate courses show page

### PR DESCRIPTION
**Pull request by goodviber**
### Context
https://trello.com/c/fK2AISht/615-migrate-courses-course-show-page-status-sidebar

The "Courses" section (linked from a page like this https://qa.publish-teacher-training-courses.service.gov.uk/organisations/2GU/2022) needs to be reimplemented in New Publish.

What needs to be done
Migrate the status side bar (the withdrawing functionality is in a follow up ticket)

### Changes proposed in this pull request
Add description_status_panel partial to description_content
### Guidance to review

There seem to be 3 routes missing referenced by the partial, 
withdraw_provider_recruitment_cycle_course_path
delete_provider_recruitment_cycle_course_path
preview_provider_recruitment_cycle_course_path

Could be more conditional testing perhaps?
### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased master
- [ ] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

File: app/views/publish/courses/_description_status_panel.html.erb [hunk 1]
```diff
@@ -0,0 +1,39 @@
+<div class="app-status-box">
+  <h3 class="govuk-heading-m">Status</h3>
+  <p class="govuk-body" data-qa="course__content-status">
+    <%= course.status_tag %>
```
[the remark this answers is not in the document]
Comment (goodviber) [reply app/views/publish/courses/_description_status_panel.html.erb:4]: Isn't that covered in teacher-training-api/app/decorators/application_decorator.rb ?

[the remark this answers is not in the document]
Comment (goodviber) [reply app/views/publish/courses/_description_status_panel.html.erb:4]: image

**Review by defong — CHANGES_REQUESTED** [review]
image

https://teacher-training-api-pr-2530.london.cloudapps.digital/publish/organisations/2X3/2022/courses/A363

vs

https://qa.publish-teacher-training-courses.service.gov.uk/organisations/2X3/2022/courses/A363


There seems to be a mismatch of what should be displayed

File: app/views/publish/courses/_description_status_panel.html.erb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  </p>
+  <h3 class="govuk-heading-s govuk-!-margin-bottom-0">
+    <% if @recruitment_cycle.current_and_open? %>Is it<% else %>Will it be<% end %> on <abbr class="app-!-text-decoration-underline-dotted" title="Find postgraduate teacher training">Find</abbr>?
+  </h3>
+  <p class="govuk-body" data-qa="course__is_findable">
+    <%= course.on_find %>
+  </p>
+
+  <% if course.has_unpublished_changes? && course.is_running? %>
```
Comment (goodviber) [inline app/views/publish/courses/_description_status_panel.html.erb:13]: does this need the `is_running?`

**Review by defong — CHANGES_REQUESTED** [review]
image

https://teacher-training-api-pr-2530.london.cloudapps.digital/publish/organisations/2AT/2022/courses/2T89
vs
https://qa.publish-teacher-training-courses.service.gov.uk/organisations/2AT/2022/courses/2T89

File: spec/models/course_spec.rb [hunk 3]
```diff
@@ -2414,7 +2414,7 @@
       let(:enrichment) { create(:course_enrichment, :subsequent_draft) }
       let(:course) { create(:course, enrichments: [enrichment]) }
 
-      it "returns true" do
+      it "returns false" do
```
[the remark this answers is not in the document]
Comment (goodviber) [reply spec/models/course_spec.rb:2417]: 2 true...

File: spec/features/publish/viewing_a_course_spec.rb [hunk 4]
```diff
@@ -15,6 +15,7 @@
       given_i_am_authenticated_as_a_provider_user(course: build(:course, enrichments: [course_enrichment], funding_type: "fee"))
       when_i_visit_the_course_page
       then_i_should_see_the_description_of_the_fee_course
+      and_i_should_see_the_status_sidebar
```
[the remark this answers is not in the document]
Comment (goodviber) [reply spec/features/publish/viewing_a_course_spec.rb:18]: Coolio

File: spec/features/publish/viewing_a_course_spec.rb [hunk 5]
```diff
@@ -23,9 +24,42 @@
...
+    expect(publish_provider_courses_show_page).to have_status_sidebar
+  end
+
+  def and_i_should_see_the_unpublished_partial
+    expect(publish_provider_courses_show_page).to have_unpublished_partial
+  end
+
+  def and_i_should_see_the_published_partial
+    expect(publish_provider_courses_show_page).to have_published_partial
+  end
```
Comment (defong) [inline spec/features/publish/viewing_a_course_spec.rb:61]: For completeness can you do the withdrawn partial as well please

File: spec/support/page_objects/publish/provider_courses_show.rb [hunk 6]
```diff
@@ -24,6 +24,9 @@ class ProviderCoursesShow < PageObjects::Base
       section :other_requirements, Sections::SummaryList, '[data-qa="enrichment__other_requirements"]'
 
       element :basic_details_link, "a.govuk-link.govuk-tabs__tab", text: "Basic details"
+      element :status_sidebar, '[data-qa="course__status_panel"]'
+      element :unpublished_partial, '[data-qa="unpublished__partial"]'
+      element :published_partial, '[data-qa="published__partial"]'
```
Comment (defong) [inline spec/support/page_objects/publish/provider_courses_show.rb:29]: These should be nested in `status_sidebar`

File: app/views/publish/courses/_description_status_panel.html.erb [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+  <p class="govuk-body" data-qa="course__content-status">
+    <%= course.status_tag %>
+  </p>
+  <h3 class="govuk-heading-s govuk-!-margin-bottom-0">
+    <% if @recruitment_cycle.current_and_open? %>Is it<% else %>Will it be<% end %> on <abbr class="app-!-text-decoration-underline-dotted" title="Find postgraduate teacher training">Find</abbr>?
+  </h3>
+  <p class="govuk-body" data-qa="course__is_findable">
+    <%= course.on_find %>
+  </p>
+<!-- Do we need a default here? -->
```
Comment (defong) [inline app/views/publish/courses/_description_status_panel.html.erb:12]: I think the default was the `unpublished` beforehand


https://github.com/DFE-Digital/publish-teacher-training/blob/master/app/views/courses/_status_panel.html.erb#L13-L19

File: spec/support/page_objects/publish/provider_courses_show.rb (hunk 6 above)
Comment (goodviber) [reply spec/support/page_objects/publish/provider_courses_show.rb:29]: I think they can only be nested within a `section` ? Done

File: app/views/publish/courses/_description_status_panel.html.erb (hunk 7 above)
Comment (goodviber) [reply app/views/publish/courses/_description_status_panel.html.erb:12]: Yes, but the current change is due to the logic chain in `is_published?`. We have to check `unpublished` first in the `if` block, just wondering if I am allowing a null condition to creep in? Or do we catch all possible conditions as is?

File: spec/features/publish/viewing_a_course_spec.rb (hunk 5 above)
Comment (goodviber) [reply spec/features/publish/viewing_a_course_spec.rb:61]: I don't think I can do that yet, the `withdrawn` functionality is another ticket?

File: app/views/publish/courses/_description_status_panel.html.erb (hunk 7 above)
Comment (goodviber) [reply app/views/publish/courses/_description_status_panel.html.erb]: Have set the default, looks a bit quirky but there you go...

**Review by defong — COMMENTED** [review]
Looking good, just a few housekeeping see [here](https://dfedigital.atlassian.net/wiki/spaces/BaT/pages/1968963648/Development+conventions)

**Review by defong — APPROVED** [review]

**Pull request merged**